### ISSUE TITLE
Wavelet packet 2D: H/V subbands are swapped when separable=True

### ISSUE THREAD

**Issue by NiclasPi** [v0lta/PyTorch-Wavelet-Toolbox#137]
`WaveletDetailTuple2d` own docstring says "We follow the pywt convention for the orientation of axes , i.e. axis 0 is horizontal and axis 1 vertical."

With `separable=True` this is not the case. Instead, the `H` and `V` axes are swapped.

### Test gap

Both tests (`test_2d_packets`, `test_boundary_matrix_packets2`) would catch this, but these are not parametrized over `separable`.
The tests for separable decomposition also dont catch this bug, as it does not compare against PyWavelet's node contents.

### Reproduce

**Example output from the script below:**
```plaintext
Signal varies along W (columns), constant along H  [ptwt mode='reflect', separable=False]
  pywt natural order: ['a', 'h', 'v', 'd']
  ptwt natural order: ['a', 'h', 'v', 'd']
  pywt freq order: [['a', 'v'], ['h', 'd']]
  ptwt freq order: [['a', 'v'], ['h', 'd']]
  natural order comparison:
    pos     pywt            ptwt            
    0       a       0.000   a       0.000   
    1       h       0.000   h       0.000   
    2       v      32.000   v      32.000   
    3       d       0.000   d       0.000   
  freq order comparison:
    pos     pywt            ptwt            
    (0, 0)  a       0.000   a       0.000   
    (0, 1)  v      32.000   v      32.000   
    (1, 0)  h       0.000   h       0.000   
    (1, 1)  d       0.000   d       0.000   

Signal varies along W (columns), constant along H  [ptwt mode='reflect', separable=True]
  pywt natural order: ['a', 'h', 'v', 'd']
  ptwt natural order: ['a', 'h', 'v', 'd']
  pywt freq order: [['a', 'v'], ['h', 'd']]
  ptwt freq order: [['a', 'v'], ['h', 'd']]
  natural order comparison:
    pos     pywt            ptwt            
    0       a       0.000   a       0.000   
    1       h       0.000   h      32.000   
    2       v      32.000   v       0.000   
    3       d       0.000   d       0.000   
  freq order comparison:
    pos     pywt            ptwt            
    (0, 0)  a       0.000   a       0.000   
    (0, 1)  v      32.000   v       0.000   
    (1, 0)  h       0.000   h      32.000   
    (1, 1)  d       0.000   d       0.000   

```

**Script to reproduce:**
```python
import numpy as np
import torch
import pywt
import ptwt

H, W = 8, 8
WAVELET = "haar"


def make_varies_along_w():
    # Nyquist-frequency pattern across columns, constant down each column.
    row = np.array([1, -1, 1, -1, 1, -1, 1, -1], dtype=np.float64)
    return np.tile(row, (H, 1))


def make_varies_along_h():
    # Nyquist-frequency pattern down rows, constant across each row.
    col = np.array([1, -1, 1, -1, 1, -1, 1, -1], dtype=np.float64)
    return np.tile(col.reshape(-1, 1), (1, W))


def build_pywt_wp(signal_np):
    return pywt.WaveletPacket2D(data=signal_np, wavelet=WAVELET, mode="reflect", maxlevel=1)


def build_ptwt_wp(signal_np, mode, separable):
    signal = torch.from_numpy(signal_np).to(torch.float64)
    return ptwt.WaveletPacket2D(
        data=signal.unsqueeze(0),
        wavelet=WAVELET,
        mode=mode,
        maxlevel=1,
        orthogonalization="qr",
        separable=separable,
    )


def pywt_grid(wp, order):
    """Official pywt.WaveletPacket2D, level 1, order='natural' (flat) or 'freq' (rows x cols)."""
    nodes = wp.get_level(1, order=order)
    if order == "freq":
        return {
            (r, c): (node.path, np.abs(node.data).sum())
            for r, row in enumerate(nodes)
            for c, node in enumerate(row)
        }
    return {i: (node.path, np.abs(node.data).sum()) for i, node in enumerate(nodes)}


def ptwt_grid(wp, order):
    """ptwt.WaveletPacket2D, level 1, order='natural' (flat) or 'freq' (rows x cols)."""
    nodes = wp.get_level(1, order=order)
    if order == "freq":
        return {
            (r, c): (node, wp[node].abs().sum().item())
            for r, row in enumerate(nodes)
            for c, node in enumerate(row)
        }
    return {i: (node, wp[node].abs().sum().item()) for i, node in enumerate(nodes)}


def print_grid_table(title, pywt_g, ptwt_g, positions):
    print(f"  {title}")
    print(f"    {'pos':<8}{'pywt':<16}{'ptwt':<16}")
    for pos in positions:
        pywt_label, pywt_val = pywt_g[pos]
        ptwt_label, ptwt_val = ptwt_g[pos]
        print(
            f"    {str(pos):<8}"
            f"{pywt_label:<4}{pywt_val:>9.3f}   "
            f"{ptwt_label:<4}{ptwt_val:>9.3f}   "
        )


def compare(signal_np, label, mode, separable):
    pywt_wp = build_pywt_wp(signal_np)
    ptwt_wp = build_ptwt_wp(signal_np, mode=mode, separable=separable)

    print(f"\n{label}  [ptwt mode={mode!r}, separable={separable}]")
    print("  pywt natural order:", [node.path for node in pywt_wp.get_level(1, order="natural")])
    print("  ptwt natural order:", ptwt_wp.get_level(1, order="natural"))
    print("  pywt freq order:", [[n.path for n in node] for node in pywt_wp.get_level(1, order="freq")])
    print("  ptwt freq order:", ptwt_wp.get_level(1, order="freq"))

    natural_positions = [0, 1, 2, 3]
    print_grid_table(
        "natural order comparison:",
        pywt_grid(pywt_wp, order="natural"),
        ptwt_grid(ptwt_wp, order="natural"),
        natural_positions,
    )

    freq_positions = [(0, 0), (0, 1), (1, 0), (1, 1)]
    print_grid_table(
        "freq order comparison:",
        pywt_grid(pywt_wp, order="freq"),
        ptwt_grid(ptwt_wp, order="freq"),
        freq_positions,
    )


varies_along_w = make_varies_along_w()
varies_along_h = make_varies_along_h()

for mode in ["reflect", "boundary"]:
    for separable in [False, True]:
        compare(varies_along_w, "Signal varies along W (columns), constant along H", mode, separable)
        compare(varies_along_h, "Signal varies along H (rows), constant along W", mode, separable)
```
